tests/integration/paramset_test.go

```diff
@@ -0,0 +1,239 @@
+			updatedSecondParamName := gjson.Get(jsonOut, "0.parameters.1.name").String()
+			Expect(int64(1)).To(Equal(updatedNumParamSets))
+			Expect(int64(2)).To(Equal(updatedNumParams))
+			Expect(ps.Spec.Name).To(Equal(updatedFirstName))
+			Expect(paramName).To(Equal(updatedFirstParamName))
+			Expect(updateParamName).To(Equal(updatedSecondParamName))
+
+			Log("delete a parameter set")
+			Expect(k8sClient.Delete(ctx, ps)).Should(Succeed())
+			Expect(waitForParamSetDeleted(ctx, ps)).Should(Succeed())
+
+			Log("verify parameter set is gone from porter data store")
+			delJsonOut := runAgentAction(ctx, "delete-check-parameters-list", ns, []string{"parameters", "list", "-n", ns, "-o", "json"})
+			delNumParams := gjson.Get(delJsonOut, "#").Int()
+			Expect(int64(0)).To(Equal(delNumParams))
+
+			Log("verify parameter set CRD is gone from k8s cluster")
+			nsName := apitypes.NamespacedName{Namespace: ps.Namespace, Name: ps.Name}
+			getPS := &porterv1.ParameterSet{}
+			Expect(k8sClient.Get(ctx, nsName, getPS)).ShouldNot(Succeed())
+
+		})
+	})
+})
+
+//NewTestParamSet minimal ParameterSet CRD for tests
+func NewTestParamSet(psName string) *porterv1.ParameterSet {
+	ps := &porterv1.ParameterSet{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "porter.sh/v1",
+			Kind:       "ParameterSet",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "porter-test-me-",
+		},
+		Spec: porterv1.ParameterSetSpec{
+			//TODO: get schema version from porter version
+			// https://github.com/getporter/porter/pull/2052
+			SchemaVersion: "1.0.1",
+			Name:          psName,
+		},
+	}
+	return ps
+}
+
+func waitForPorterPS(ctx context.Context, ps *porterv1.ParameterSet, msg string) error {
+	Log("%s: %s/%s", msg, ps.Namespace, ps.Name)
+	key := client.ObjectKey{Namespace: ps.Namespace, Name: ps.Name}
+	ctx, cancel := context.WithTimeout(ctx, getWaitTimeout())
+	defer cancel()
+	for {
+		select {
+		case <-ctx.Done():
+			Fail(errors.Wrapf(ctx.Err(), "timeout %s", msg).Error())
+		default:
+			err := k8sClient.Get(ctx, key, ps)
+			if err != nil {
+				// There is lag between creating and being able to retrieve, I don't understand why
+				if apierrors.IsNotFound(err) {
+					time.Sleep(time.Second)
+					continue
+				}
+				return err
+			}
+
+			// Check if the latest change has been processed
+			if ps.Generation == ps.Status.ObservedGeneration {
+				if apimeta.IsStatusConditionTrue(ps.Status.Conditions, string(porterv1.ConditionComplete)) {
+					time.Sleep(time.Second)
+					return nil
+				}
+
+				if apimeta.IsStatusConditionTrue(ps.Status.Conditions, string(porterv1.ConditionFailed)) {
+					// Grab some extra info to help with debugging
+					debugFailedPSCreate(ctx, ps)
+					return errors.New("porter did not run successfully")
+				}
+			}
+
+			time.Sleep(time.Second)
+			continue
+		}
+	}
+}
+
+func waitForParamSetDeleted(ctx context.Context, ps *porterv1.ParameterSet) error {
```

Comment:
The amount of copy/paste that is in these test files is really starting to concern me. We are just piling up on code that we know we don't want to maintain like this.

I've created https://github.com/getporter/operator/issues/102 to tack consolidating these test helpers so that 80% of the test files aren't duplicate test helpers.

Reply:
Next on the list, will be starting immediately